atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidatorTest.java

```diff
@@ -364,8 +410,56 @@ public void validateNewlyAddedHostsAddedIfAgreedWithOldHostsOnPreviousTopologyWh
     }
 
     @Test
-    public void validateNewlyAddedHostsAddedIfAgreedWithOldTopologyWhenNoQuorumAndNoCurrentServers() {
-        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(ALL_HOSTS);
+    public void validateNewlyAddedHostsAddedIfPresentingPlausiblyEvolvedPreviousTopologyWhenNoQuorum() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = initialiseHosts(ALL_HOSTS);
+        Map<CassandraServer, CassandraClientPoolingContainer> oldHosts =
+                filterServerToContainerMap(allHosts, OLD_HOSTS::contains);
+        Set<CassandraServer> oldCassandraServers = oldHosts.keySet();
+        Set<CassandraServer> newCassandraServers = filterServers(allHosts, NEW_HOSTS::contains);
+
+        setHostIds(filterContainers(allHosts, OLD_HOSTS::contains), HostIdResult.success(UUIDS));
+        validator.getNewHostsWithInconsistentTopologies(mapToTokenRangeOrigin(oldCassandraServers), oldHosts);
+
+        Set<String> hostsOffline = OLD_HOSTS;
+        setHostIds(filterContainers(allHosts, hostsOffline::contains), HostIdResult.hardFailure());
+        setHostIds(
+                filterContainers(allHosts, server -> !hostsOffline.contains(server)),
+                HostIdResult.success(MAYBE_SAME_AS_FIRST_CLUSTER));
+
+        assertThat(hostsOffline).hasSizeGreaterThanOrEqualTo((ALL_HOSTS.size() + 1) / 2);
+        assertThat(validator.getNewHostsWithInconsistentTopologies(
+                        mapToTokenRangeOrigin(newCassandraServers), allHosts))
+                .as("accepts quorum from new hosts if they have at least one host IDs from the old topology")
+                .isEmpty();
+    }
+
+    @Test
+    public void validateNewlyAddedHostsAddedIfAgreedWithOldTopologyWhenNoQuorumAndNoCurrentServersWithRingOrigin() {
```

Comment:
nit: This is both a mouthful and not sufficiently descriptive - it works because the ring servers match the config host names! I don't have a great alternative, and given time, this can be ignored (and it'll probably get changed soon enough anyway...)